src/Bridge/Symfony/Messenger/DataPersister.php

```diff
@@ -98,4 +86,33 @@ public function remove($data, array $context = [])
                 ->with(new RemoveStamp())
         );
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function loop($data, array $context = []): bool
+    {
+        $value = $this->getMessengerAttributeValue($this->resourceMetadataFactory->create($context['resource_class'] ?? $this->getObjectClass($data)), $context);
+
+        return 'persist' === $value || (\is_array($value) && \in_array('persist', $value, true));
+    }
+
+    private function getMessengerAttributeValue(ResourceMetadata $resourceMetadata, array $context = [])
+    {
+        if (null !== $operationName = $context['collection_operation_name'] ?? $context['item_operation_name'] ?? null) {
+            return $resourceMetadata->getTypedOperationAttribute(
+                    $context['collection_operation_name'] ?? false ? OperationType::COLLECTION : OperationType::ITEM,
+                    $operationName,
+                    'messenger',
+                    false,
+                    true
+                );
+        }
+
+        if (isset($context['graphql_operation_name'])) {
+            return $resourceMetadata->getGraphqlAttribute($context['graphql_operation_name'], 'messenger', false, true);
+        }
+
+        return $resourceMetadata->getAttribute('messenger', false);
```

Comment:
I think we can safely return `false` here as we're already using the `resourceFallback` in the getAttributes methods above.